File: moment/moment.go
package moment

import (
	"time"
)

type Moment struct {
	Date  time.Time `yaml:"date"`
	Title string    `yaml:"title"`
	Text  string    `yaml:"text"`
}

func (m *Moment) Path() string {
	return m.Date.Format("/2006/01/02/15:04:05/")
}

File: backend/src/main/kotlin/net/hexe/the_gear_of_seasons/shelf/Moment.kt
package net.hexe.the_gear_of_seasons.shelf

import java.time.LocalDateTime
import java.time.ZoneId
import java.util.*

class Moment {
  var date: Date = Date(0)
  var title = ""
  var author = ""
  var text = ""

  fun localTime(): LocalDateTime = LocalDateTime.ofInstant(date.toInstant(), ZoneId.of("Asia/Tokyo"))
}
